cdk/kittyhawk/src/application.ts

```diff
@@ -40,24 +40,26 @@ export class Application extends Construct {
  * @param envKey name of the environment variable to insert
  * @param envValue value of the environment variable
  */
-function insertIfNotPresent(envArray: { name: string; value: string }[], envKey: string, envValue: any) {
-  const envSettingsModule = envArray?.filter(env => (env.name === envKey));
-  if (envSettingsModule?.length > 0) {
+export function insertIfNotPresent(envArray: { name: string; value: string }[], envKey: string, envValue: any) {
+  const envSettingsModule = envArray.filter(env => (env.name === envKey));
+  if (envSettingsModule.length > 0) {
     throw new Error(`${envKey} should not be redefined as an enviroment variable.`);
   }
   envArray.push({ name: envKey, value: envValue });
 
 }
 export interface DjangoApplicationProps extends ApplicationProps {
   /**
-   * Domain of the application.
+   * List of domain(s) of the application and if they are a subdomain.
    */
-  readonly domain: string;
+  readonly domains: {host: string; isSubdomain: boolean}[];
 
   /**
-   * Just the list of paths passed to the ingress since we already know the host.
+   * Just the list of paths passed to the ingress since we already know the host. Optional.
+   *
+   * @default undefined
    */
-  readonly ingressPaths: string[];
+  readonly ingressPaths?: string[];
```

Comment:
Shouldn't this still be required so we always know what paths should route to django?

Reply:
optional rn since i changed djangoapplication to support celery, which doesn't need ingress or ingressPaths.

Reply:
Ah, that would make sense, that's my bad.